agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -373,7 +373,13 @@ func (dg *dockerGoClient) pullImage(image string, authData *apicontainer.Registr
 		return CannotGetDockerClientError{version: dg.version, err: err}
 	}
 
-	authConfig, err := dg.getAuthdata(image, authData)
+	sdkAuthConfig, err := dg.getAuthdata(image, authData)
+	authConfig := docker.AuthConfiguration{
```

Comment:
Could you please add a `TODO` here that we won't need `docker.AuthConfiguration` anymore when we migrate to SDK's pull image?